cylc/flow/prerequisite.py

```diff
@@ -79,6 +75,22 @@ def __init__(self, point, start_point=None):
         # * `False` (prerequisite unsatisfied).
         self._all_satisfied = None
 
+    def __eq__(self, other):
+        """Test equality of prerequisites. Ignore derived attributes."""
+        return self.hash() == other.hash()
+
```

Comment:
There's a problem with this, but it isn't actually used so might as well remove?
```suggestion
```

Reply:
What's the problem?

Reply:
Should be `.instantaneous_hash()` and also it doesn't account for testing equality against non-`Prerequisite` objects (although this is unlikely to happen)

Reply:
You're right that it isn't used now (it was when using the list-based duplicate checking) so it could be removed.

Happy for you and Oliver to take over the branch and tweak as needed to get it in!

Reply:
(oops, I accidentally hit commit on that - but it needed to be done anyway)